tests/Http/PostTest.php

```diff
@@ -762,6 +762,43 @@ public function testPostsIndexAsNonAdminNonOwner()
         ]);
     }
 
+    /**
+     * Test for retrieving all posts with some anonymous posts.
+     * Non-owners should not northstar_id or location.
+     *
+     * GET /api/v3/posts
+     * @return void
+     */
+    public function testPostsIndexWithAnonymousPosts()
+    {
+        // Create an accepted post.
+        $regularPost = factory(Post::class, 'accepted')->create();
+
+        // Create 1 post that is anonymous.
+        sleep(1);
+        $anonymousPost = factory(Post::class, 'accepted')->create();
+        $anonymousPost->actionModel->anonymous = 1;
+        $anonymousPost->actionModel->save();
+
+        // Anonymously hit the endpoint and norhtstar_id should not be returned for the anonymous post (first post since it was most recently created).
+        $response = $this->getJson('api/v3/posts');
+        $response->assertStatus(200);
+        $this->assertArrayNotHasKey('northstar_id', $response->decodeResponseJson()['data'][0]);
+        $this->assertEquals($regularPost->northstar_id, $response->decodeResponseJson()['data'][1]['northstar_id']);
+
+        // Hit the endpoint with admin credientals and same results as above.
+        $response = $this->withAdminAccessToken()->getJson('api/v3/posts');
+        $response->assertStatus(200);
+        $this->assertArrayNotHasKey('northstar_id', $response->decodeResponseJson()['data'][0]);
```

Comment:
Hmm, admins should be able to view Northstar IDs for these posts. Let's double-check the permission logic.

Reply:
updated this test to instead check that another logged in user cannot see `northstar_id` and adding another line to assure that admins can still see this!